editors/code/src/main.ts

```diff
@@ -109,3 +99,107 @@ export async function deactivate() {
     await ctx?.client?.stop();
     ctx = undefined;
 }
+
+async function bootstrap(config: Config, state: PersistentState): Promise<string> {
+    await bootstrapExtension(config, state);
+    const path = await bootstrapServer(config, state);
+    return path;
+}
+
+async function bootstrapExtension(config: Config, state: PersistentState): Promise<void> {
+    // TODO: ask to install manually if we are nightly.
+    if (config.channel == "stable") return;
+    // TODO: don't do this more often than once an hour.
+    const shouldDownloadNightly = state.releaseId == undefined || true;
+    if (!shouldDownloadNightly) return;
+
+    let release = await fetchRelease("nightly");
+    if (release.id == state.releaseId) return;
+
+    const artifact = release.assets.find(artifact => artifact.name === "rust-analyzer.vsix");
+    assert(!!artifact, `Bad release: ${JSON.stringify(release)}`);
+
+    const dest = path.join(config.globalStoragePath, "rust-analyzer.vsix");
+    await download(artifact.browser_download_url, dest, "Downloading rust-analyzer extension");
+
+    await await vscode.commands.executeCommand("workbench.extensions.installExtension", vscode.Uri.file(dest));
+    await fs.unlink(dest);
+
+    await state.updateReleaseId(release.id);
+    await vscode.commands.executeCommand("workbench.action.reloadWindow");
+    assert(false, "unreachable");
+}
+
+async function bootstrapServer(config: Config, state: PersistentState): Promise<string> {
+    let path = await getServer(config, state);
+    if (!path) {
+        throw new Error(
+            "Rust Analyzer Language Server is not available. " +
+            "Please, ensure its [proper installation](https://rust-analyzer.github.io/manual.html#installation)."
+        );
+    }
+
+    const res = spawnSync(path, ["--version"]);
+    log.debug("Checked binary availability via --version", res);
+    log.debug(res, "--version output:", res.output?.map(String));
```

Comment:
Please preserve the comment about the declared type inconsistency

Reply:
just made the type correct by adding an explicit encoding.